.github/workflows/build.yml

```diff
@@ -0,0 +1,28 @@
+name: Debug Build
+on:
+  pull_request:
+  push:
```

Comment:
I think building on PRs would be enough. If it is pushed, then it should have been verified once already in the PR.